weed/util/fullpath.go

```diff
@@ -1,19 +1,23 @@
 package util
 
 import (
-	"path/filepath"
+	"path"
 	"strings"
 )
 
+// FullPath will keep the literal passed in.
+// file literal: /path/to/file
+// directory literal: /path/
+// change to file literal: fp.Child(".") or NewFullPath(name, ".")
+// change to directory literal: fp.Child("./") or NewFullPath(name, "./")
 type FullPath string
 
-func NewFullPath(dir, name string) FullPath {
-	return FullPath(dir).Child(name)
+func NewFullPath(name ...string) FullPath {
```

Comment:
Why need to support array here? This was not intended as a replacement for Join.